xilem/src/view/board.rs

```diff
@@ -0,0 +1,384 @@
+        BoardElement {
+            element: child.inner.svg_boxed().into(),
+        }
+    }
+
+    fn with_downcast_val<R>(
+        mut this: Mut<'_, Self>,
+        f: impl FnOnce(Mut<'_, Pod<PositionedElement<W>>>) -> R,
+    ) -> (Self::Mut<'_>, R) {
+        let r = {
+            let mut child = this.parent.child_mut(this.idx);
+            f(child.downcast())
+        };
+        (this, r)
+    }
+}
+
+impl<W: Widget> AnyElement<Pod<PositionedElement<W>>> for BoardElement {
+    fn replace_inner(mut this: Self::Mut<'_>, child: Pod<PositionedElement<W>>) -> Self::Mut<'_> {
+        this.parent.remove_child(this.idx);
+        this.parent.insert_child(this.idx, child.inner.svg_boxed());
+        this
+    }
+}
+
+impl SuperElement<Pod<KurboShape>> for BoardElement {
+    fn upcast(child: Pod<KurboShape>) -> Self {
+        BoardElement {
+            element: child.inner.svg_boxed().into(),
+        }
+    }
+
+    fn with_downcast_val<R>(
+        mut this: Mut<'_, Self>,
+        f: impl FnOnce(Mut<'_, Pod<KurboShape>>) -> R,
+    ) -> (Self::Mut<'_>, R) {
+        let r = {
+            let mut child = this.parent.child_mut(this.idx);
+            f(child.downcast())
+        };
+        (this, r)
+    }
+}
+
+impl AnyElement<Pod<KurboShape>> for BoardElement {
+    fn replace_inner(mut this: Self::Mut<'_>, child: Pod<KurboShape>) -> Self::Mut<'_> {
+        this.parent.remove_child(this.idx);
+        this.parent.insert_child(this.idx, child.inner.svg_boxed());
+        this
+    }
+}
+
+impl ElementSplice<BoardElement> for BoardSplice<'_> {
+    fn insert(&mut self, BoardElement { element }: BoardElement) {
+        self.element.insert_child(self.idx, element.inner);
+        self.idx += 1;
+    }
+
+    fn with_scratch<R>(&mut self, f: impl FnOnce(&mut AppendVec<BoardElement>) -> R) -> R {
+        let ret = f(&mut self.scratch);
+        for BoardElement { element } in self.scratch.drain() {
+            self.element.insert_child(self.idx, element.inner);
+            self.idx += 1;
+        }
+        ret
+    }
+
+    fn mutate<R>(&mut self, f: impl FnOnce(Mut<'_, BoardElement>) -> R) -> R {
+        let child = BoardElementMut {
+            parent: self.element.reborrow_mut(),
+            idx: self.idx,
+        };
+        let ret = f(child);
+        self.idx += 1;
+        ret
+    }
+
+    fn delete<R>(&mut self, f: impl FnOnce(Mut<'_, BoardElement>) -> R) -> R {
+        let ret = {
+            let child = BoardElementMut {
+                parent: self.element.reborrow_mut(),
+                idx: self.idx,
+            };
+            f(child)
+        };
+        self.element.remove_child(self.idx);
+        ret
+    }
+
+    fn skip(&mut self, n: usize) {
+        self.idx += n;
+    }
+}
+
+/// An ordered sequence of views for a [`Board`] view.
+/// See [`ViewSequence`] for more technical details.
+pub trait BoardSequence<State, Action = ()>:
+    ViewSequence<State, Action, ViewCtx, BoardElement>
```

Comment:
Same here, this could be with the suggestions above:
```suggestion
pub trait SvgSequence<State, Action = ()>:
    ViewSequence<State, Action, ViewCtx, Pod<Box<dyn SvgElement>>>
```